client/client.go

```diff
@@ -401,3 +424,26 @@ func (r *redirectedHTTPAction) HTTPRequest(ep url.URL) *http.Request {
 	orig.URL = &r.location
 	return orig
 }
+
+// randPermStartWithX returns, as a slice of n ints, a pseudo-random permutation
+// of the integers [0,n), which starts with x.
+func randPermStartWithX(r *rand.Rand, n, x int) []int {
```

Comment:
this seems to be over complicated. I think we can get the perm and append it to the pined one. (yea... the pined one might appear twice, but it does not really matter)